pkg/storage/streams/store.go

```diff
@@ -60,16 +64,32 @@ type Store interface {
 
 // streamStore is a store for streams
 type streamStore struct {
-	segments    segments.Store
-	segmentSize int64
+	segments     segments.Store
+	segmentSize  int64
+	rootKey      []byte
+	encBlockSize int
+	encType      int
 }
 
 // NewStreamStore stuff
-func NewStreamStore(segments segments.Store, segmentSize int64) (Store, error) {
+func NewStreamStore(segments segments.Store, segmentSize int64, rootKey string, encBlockSize int, encType int) (Store, error) {
```

Comment:
Why is the `rootKey` here a string, similar question for `encType`.

Reply:
This is because they are passed in directly from the config in miniogw. As cli arguments they are string/int instead of Key/Cipher. We can do the casting in miniogw if you think that's better than doing it from within streamstore.